Mathlib/LinearAlgebra/Multilinear/Basis.lean

```diff
@@ -11,50 +11,27 @@ import Mathlib.LinearAlgebra.Multilinear.Curry
 
 This file proves lemmas about the action of multilinear maps on basis vectors.
 
-## TODO
-
- * Refactor the proofs in terms of bases of tensor products, once there is an equivalent of
-   `Basis.tensorProduct` for `PiTensorProduct`.
-
 -/
 
 
 open MultilinearMap
 
-variable {R : Type*} {ι : Type*} {n : ℕ} {M : Fin n → Type*} {M₂ : Type*} {M₃ : Type*}
-variable [CommSemiring R] [AddCommMonoid M₂] [AddCommMonoid M₃] [∀ i, AddCommMonoid (M i)]
-variable [∀ i, Module R (M i)] [Module R M₂] [Module R M₃]
-
-/-- Two multilinear maps indexed by `Fin n` are equal if they are equal when all arguments are
-basis vectors. -/
-theorem Basis.ext_multilinear_fin {f g : MultilinearMap R M M₂} {ι₁ : Fin n → Type*}
-    (e : ∀ i, Basis (ι₁ i) R (M i))
-    (h : ∀ v : ∀ i, ι₁ i, (f fun i => e i (v i)) = g fun i => e i (v i)) : f = g := by
-  induction n with
-  | zero =>
-    ext x
-    convert h finZeroElim
-  | succ m hm =>
-    apply Function.LeftInverse.injective uncurry_curryLeft
-    refine Basis.ext (e 0) ?_
-    intro i
-    apply hm (Fin.tail e)
-    intro j
-    convert h (Fin.cons i j)
-    iterate 2
-      rw [curryLeft_apply]
-      congr 1 with x
-      refine Fin.cases rfl (fun x => ?_) x
-      dsimp [Fin.tail]
-      rw [Fin.cons_succ, Fin.cons_succ]
+variable {ι R : Type*} [CommSemiring R]
+  {M : ι → Type*} [∀ i, AddCommMonoid (M i)] [∀ i, Module R (M i)]
+  {N : Type*} [AddCommMonoid N] [Module R N]
 
 /-- Two multilinear maps indexed by a `Fintype` are equal if they are equal when all arguments
-are basis vectors. Unlike `Basis.ext_multilinear_fin`, this only uses a single basis; a
-dependently-typed version would still be true, but the proof would need a dependently-typed
-version of `dom_dom_congr`. -/
-theorem Basis.ext_multilinear [Finite ι] {f g : MultilinearMap R (fun _ : ι => M₂) M₃} {ι₁ : Type*}
-    (e : Basis ι₁ R M₂) (h : ∀ v : ι → ι₁, (f fun i => e (v i)) = g fun i => e (v i)) : f = g := by
+are basis vectors. -/
+theorem Basis.ext_multilinear [Finite ι] {f g : MultilinearMap R M N} {ιM : ι → Type*}
+    (e : ∀ i, Basis (ιM i) R (M i))
+    (h : ∀ v : (i : ι) → ιM i, (f fun i ↦ e i (v i)) = g fun i ↦ e i (v i)) : f = g := by
   cases nonempty_fintype ι
-  exact
-    (domDomCongr_eq_iff (Fintype.equivFin ι) f g).mp
-      (Basis.ext_multilinear_fin (fun _ => e) fun i => h (i ∘ _))
+  classical
+  ext m
+  rcases Function.Surjective.piMap (fun i ↦  (e i).repr.symm.surjective) m with ⟨x, rfl⟩
+  unfold Pi.map
+  simp only [(e _).repr_symm_apply, Finsupp.linearCombination_apply, Finsupp.sum,
```

Comment:
Is un-squeezing slower? If not, can you do so, please?

Reply:
None of these lemmas are in the default `simp` set, so I'm replacing with `simp_rw`.